src/Select.js

```diff
@@ -177,7 +177,7 @@ var Select = React.createClass({
 					newProps.placeholder)
 				);
 			};
-			if (this.props.asyncOptions) {
+			if (this.props.asyncOptions && this.props.autoload) {
```

Comment:
should this be ` && (this.props.autoload || newProps.value)`?

If a value is passed, we do want the async options to load, otherwise the label for the value won't be expanded correctly